runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1515,6 +1508,26 @@ TR_J9ServerVM::getROMMethodFromRAMMethod(J9Method *ramMethod)
    return std::get<0>(stream->read<J9ROMMethod *>());
    }
 
+TR_OpaqueClassBlock *
+TR_J9ServerVM::getCacheClassFromSignature(const char * sig, int32_t sigLength, bool isVettedForAOT, TR_OpaqueMethodBlock *methodBlock, J9ClassLoader * cl, bool cacheInvalidatedClass)
+   {
+   ClassLoaderStringPair key = {cl, std::string(sig, sigLength)};
+      {
+         OMR::CriticalSection getCustomClassMap(_compInfoPT->getCustomClassMapMonitor());
```

Comment:
The code is accessing the 'per compilation' cache which is only accessed by one compilation thread, therefore the monitor is not needed.